### PR TITLE
Update HttpURLConnections to follow HTTPS redirects

### PR DESCRIPTION
**Pull request by darcywong00**
For the first portion of resolving #370 

Fix KMManager and JSONParser HttpURLConnection calls so HTTP:// to HTTPS:// redirects are properly followed.

### CODE REVIEW

File: android/KMEA/app/src/main/java/com/tavultesoft/kmea/JSONParser.java [hunk 1]
```diff
@@ -30,11 +30,25 @@ public JSONObject getJSONObjectFromUrl(String urlStr) {
     HttpURLConnection urlConnection = null;
 
     try {
+      // TODO: Refactor urlConnection to utility method
       URL url = new URL(urlStr);
+      HttpURLConnection.setFollowRedirects(false);
```
Comment (jahorton) [inline android/KMEA/app/src/main/java/com/tavultesoft/kmea/JSONParser.java]: I take it that using `setFollowRedirects(true)` doesn't follow properly, hence our doing it manually in code below?

Comment (darcywong00) [reply android/KMEA/app/src/main/java/com/tavultesoft/kmea/JSONParser.java]: From what I'm reading, `setFollowRedirects` defaults to `true`, but only follows if the protocol remains the same.

**Comment by darcywong00** [comment]
1. I can refactor it into a new utility class.
2. Arbitrarily, I'll have it look up to 5 tries.

**Comment by mcdurdin** [comment]
According to https://stackoverflow.com/questions/6852846/browser-redirect-loop-detection#comment12301893_6852931:
> Firefox 10 gives up after 21 redirections as well. According to rfc2616 10.3 older versions of the HTTP protocol recommended a 5 redirection limit, so it is not safe to assume the browser will handle more than that.

However, most modern browsers do at least 20 hops: https://stackoverflow.com/a/36041063/1836776

**Pull request merged**